src/lib/utils.jl

```diff
@@ -102,12 +102,170 @@ end
 """
     hessian(f, x)
 
-Construct the Hessian of `f`, where `x` is a real or real array and `f(x)` is
-a real.
+Construct the Hessian `∂²f/∂x∂x`, where `x` is a real number or an array,
+and `f(x)` is a real number.
```

Comment:
should this say something on how it handles higher-order arrays?

Reply:
Something like the output being the shape of `vec(x) .* vec(x)'`? I did put an example marked "uses linear indexing" but could add more words too.

Reply:
yes, `vec(y) .* vec(x)'` would make the ordering very clear

Reply:
See what you think of the current wording, updated in https://github.com/FluxML/Zygote.jl/pull/890/commits/e320b0902e8faa7322da8e52d94bf9dfb21be2ab. This one is a square matrix, but Jacobian has `y`'s rows.